From fc5816a64f4b863fdce52694d35475e8ba0fb81f Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Fri, 18 Nov 2016 15:07:12 -0800
Subject: [PATCH] OS-5798 imgadm should allow "+" in an image manifest
 "version", per semver Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
 Approved by: Todd Whiteman <todd.whiteman@joyent.com>

---
 src/img/CHANGES.md                                  | 4 ++++
 src/img/node_modules/imgmanifest/lib/imgmanifest.js | 4 ++--
 src/img/node_modules/imgmanifest/package.json       | 2 +-
 src/img/package.json                                | 4 ++--
 4 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/src/img/CHANGES.md b/src/img/CHANGES.md
index b2b8f589..d2b25b90 100644
--- a/src/img/CHANGES.md
+++ b/src/img/CHANGES.md
@@ -6,6 +6,10 @@ Known issues:
   Docker Registry v1, which is now deprecated.
 
 
+## 3.7.0
+
+- OS-5798 Allow '+' in the image manifest "version" field.
+
 ## 3.6.2
 
 - OS-5335 docker image import should disallow meaningless operations
diff --git a/src/img/node_modules/imgmanifest/lib/imgmanifest.js b/src/img/node_modules/imgmanifest/lib/imgmanifest.js
index 2aa26881..e7adc897 100644
--- a/src/img/node_modules/imgmanifest/lib/imgmanifest.js
+++ b/src/img/node_modules/imgmanifest/lib/imgmanifest.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 /*
@@ -29,7 +29,7 @@ var V = 2;
 // Regexes
 var UUID_RE = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/;
 var NAME_RE = /^[A-Za-z0-9._/ -]+$/;
-var VERSION_RE = /^[A-Za-z0-9._/-]+$/;
+var VERSION_RE = /^[A-Za-z0-9._/+-]+$/;
 // published_at (ISO 8601 date string, e.g. "2012-12-25T12:00:00.123Z")
 // Required if activated.
 var PUBLISHED_AT_RE = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d{3})?Z$/;
diff --git a/src/img/node_modules/imgmanifest/package.json b/src/img/node_modules/imgmanifest/package.json
index 24d15876..7d3aa41f 100644
--- a/src/img/node_modules/imgmanifest/package.json
+++ b/src/img/node_modules/imgmanifest/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgmanifest",
   "description": "Library for working with Triton/SDC/SmartOS image manifests",
-  "version": "2.2.0",
+  "version": "2.3.0",
   "author": {
     "name": "Joyent",
     "url": "joyent.com"
diff --git a/src/img/package.json b/src/img/package.json
index d41dcef5..7cdad364 100644
--- a/src/img/package.json
+++ b/src/img/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgadm",
   "description": "Manage SmartOS virtual machine images.",
-  "version": "3.6.2",
+  "version": "3.7.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
@@ -13,7 +13,7 @@
     "docker-registry-client": "git://github.com/joyent/node-docker-registry-client.git#765e74c",
     "extsprintf": "1.2.0",
     "findit2": "2.2.3",
-    "imgmanifest": "git://github.com/joyent/node-imgmanifest.git#4f80d8c",
+    "imgmanifest": "2.3.0",
     "mkdirp": "0.5.0",
     "node-uuid": "1.4.1",
     "once": "1.3.1",
-- 
2.21.0

